File: ideal-core/src/main/java/cn/idealframework2/cache/CacheBuilderFactory.java
package cn.idealframework2.cache;

import cn.idealframework2.cache.serialize.ValueSerializer;

import javax.annotation.Nonnull;

/**
 * @author 宋志宗 on 2022/12/5
 */
public interface CacheBuilderFactory {

  @Nonnull
  <K, V> CacheBuilder<K, V> newBuilder(@Nonnull ValueSerializer<V> valueSerializer);
}

File: ideal-core/src/main/kotlin/cn/idealframework2/cache/coroutine/CacheBuilderFactory.kt
package cn.idealframework2.cache.coroutine

import cn.idealframework2.cache.serialize.ValueSerializer

/**
 * @author 宋志宗 on 2022/12/5
 */
interface CacheBuilderFactory {

  fun <K : Any, V : Any> newBuilder(valueSerializer: ValueSerializer<V>): CacheBuilder<K, V>
}
